akka-docs/src/main/paradox/typed/cluster-sharding.md

```diff
@@ -105,7 +105,11 @@ Scala
 Java
 :  @@snip [ShardingCompileOnlyTest.java](/akka-cluster-sharding-typed/src/test/java/jdocs/akka/cluster/sharding/typed/ShardingCompileOnlyTest.java) { #roles }
 
+### A note about EntityRef and serialization
 
+If including `EntityRef`s in messages or the `State`/`Event`s of an `EventSourcedBehavior`, those `EntityRef`s will need to be serialized.
+The @scala[`entityId`, `typeKey`, and (in multi-DC use-cases) `dataCenter` of an `EntityRef`]@java[`getEntityId`, `getTypeKey`, and (in multi-DC use-cases) `getDataCenter` methods of an `EntityRef`]
+provide exactly the information needed upon deserialization to regenerate an `EntityRef` equivalent to the one serialized.
```

Comment:
Clarify that there is currently no automatic serialization of EntityRef included in Akka.

I would have expect that this PR actually included serialization support also, but maybe that is a second step?
We should then have:
* support in Jackson serialization for embedded EntityRef
* a Protobuf serializer so that en EntityRef can be sent as a message (probably not that useful but for completeness and corresponding to what is supported for an ActorRef)

Reply:
> I would have expect that this PR actually included serialization support also, but maybe that is a second step?

That was because of me saying we shouldn't provide serialization out of the box for it, only make it possible for users to write such a serializer, which was not previously possible.

Reply:
that isn't trivial in Jackson

Reply:
My point of view wasn't so much about how easy it is as it was about to what extent we should encourage using entity ref as something you send around in messages and across cluster nodes vs something local you use to interact with sharding.

Reply:
I'm fine with just adding this information to the EntityRef and work on additional serialization support separately if that is considered useful.

discussed earlier at https://github.com/akka/akka/pull/27077#discussion_r299908796

Reply:
another doc section that might need an update is https://doc.akka.io/docs/akka/current/typed/interaction-patterns.html#responding-to-a-sharded-actor

Reply:
I think any deserialization is going to require some means to get a tag for the appropriate command type.  One option is an `EntityTypeTag` registry, but an alternative (perhaps only applicable for cases with a small number of entities) is:

```
object FooEntity {
  val EntityType = "FOO"
  val TypeKey = EntityTypeKey[Command](EntityType)

  def refFor(system: ActorSystem[Nothing], id: String): EntityRef[Command] =
    ClusterSharding(system).entityRefFor(TypeKey, id)
}
```

In which case the deserializer can more quickly check "is the type key `FOO`?  Then pass the id to `FooEntity.refFor`"

I actually think that the case for serialization is stronger for persistence than it is for messages, in view of the resistance to ephemeral state in `EventSourcedBehavior` (I forget the ticket number).